### PR TITLE
test: Add corner case tests for bigint delegation in unrolled circuit

### PR DESCRIPTION
**Pull request by Fitznik**
## What ❔

Added 196 corner case tests for the bigint_with_control delegation circuit in the unrolled proving pipeline. 

## Is this a breaking change?
- [ ] Yes
- [x] No

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [x] Code has been formatted.

### CODE REVIEW

File: examples/bigint_with_control/src/main.rs [hunk 1]
```diff
@@ -16,41 +17,1446 @@ unsafe extern "C" fn start_rust() -> ! {
     main()
 }
 
-#[repr(C, align(32))]
-struct AlignedU256 {
-    words: [u32; 8],
+#[inline(always)]
```
Comment (popzxc) [inline examples/bigint_with_control/src/main.rs]: Why do we need to copy-paste it from common constants? IMHO less duplication is better.

File: examples/bigint_with_control/src/main.rs [hunk 2]
```diff
@@ -16,41 +17,1446 @@ unsafe extern "C" fn start_rust() -> ! {
...
-    zksync_os_finish_success(&[
-        accumulator.words[0],
-        accumulator.words[1],
-        accumulator.words[2],
-        carry,
-        0,
-        0,
-        0,
-        0,
+    fn run_case(a: &mut U256, b: &U256, out: &mut [u32], mut mask: u32) {
```
Comment (popzxc) [inline examples/bigint_with_control/src/main.rs]: Can we create a module e.g. `test_suite.rs`, encapsulate each test case into a separate function with doc-comment, and provide a method e.g. `test_suite::run_edge_case_tests()` which will be called here? It'll be much more clear.

Also, I believe that each test case can be expressed as a set of inputs in a test vector, e.g.

```rust
struct TestInput { .. }
type ExpectedOutput = U256;

const TEST_VECTORS: [(TestInput, ExpecteOutput); ..] = ...;

fn run_edge_case_tests() {
  for (input, expected_output) in TEST_VECTORS { .. }
}
```

File: examples/bigint_with_control/Cargo.toml [hunk 3]
```diff
@@ -10,7 +10,7 @@ keywords = ["blockchain", "zksync", "zk", "risc-v"]
 categories = ["cryptography"]
 
 [dependencies]
-common_constants = { path = "../../common_constants" }
+common_constants = { path = "../../common_constants"}
```
Comment (popzxc) [inline examples/bigint_with_control/Cargo.toml:13]: Nit: I'd suggest undoing this change -- space between `}` is idiomatic

File: examples/bigint_with_control/src/test_suite.rs [hunk 4]
```diff
@@ -0,0 +1,1508 @@
...
+///
+/// Operation masks are built from bit indices defined in `common_constants`:
+///   - bit 0: ADD
+///   - bit 1: SUB
+///   - bit 2: SUB_AND_NEGATE (computes b - a)
+///   - bit 3: MUL_LOW
+///   - bit 4: MUL_HIGH
+///   - bit 5: EQ
+///   - bit 6: CARRY / BORROW-IN
+///   - bit 7: MEMCOPY
```
Comment (popzxc) [inline examples/bigint_with_control/src/test_suite.rs]: Nit: this should be `//!`, not `///` -- `///` only adds doc-comment for the item below (e.g. to this import), while `//!` adds comment to the outer item (e.g. module).

**Review by popzxc — COMMENTED** [review]
Mostly LGTM, one small nit 😅

**Review by popzxc — APPROVED** [review]
Thanks!

**Review by shamatar — APPROVED** [review]
